### DISCUSSION TITLE
xView training and result reproduction

### DISCUSSION THREAD

**Discussion by nalemadi** [fcakyon/small-object-detection-benchmark#28] [Q&A]
Hello @fcakyon,
Congratulations on your achievement and thank you for sharing the great job.

I was able to reproduce the results announced with Visdrone dataset by converting the dataset to coco format, training, and inference, which is excellent. 
But I have faced some issues with training xView with the TOOD detector. 

These are the steps I followed to train xView (similar to the actions performed on training Visdrone): 

1. I converted xView dataset to coco using this script https://github.com/fcakyon/small-object-detection-benchmark/blob/main/xview/xview_to_coco.py which produces train.json (634 images) and val.json (212 images).
2. I used https://github.com/fcakyon/small-object-detection-benchmark/blob/main/xview/slice_xview.py to slice the val split images, which makes: val_300_0.json, val_300_025.json, val_400_0.json, val_400_025.json, val_500_0.json, val_500_025.json with its corresponding images directories.
3. I used this script to train the model using xView https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_tools/train.py with this config file https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/xview_tood/tood_crop_300_500_cls_60.py (I only changed the `BATCH_MULTIPLIER` to lower value to fit into the available resources.
I had the following error:
<img width="1724" alt="Screenshot 2023-03-29 at 11 17 28 AM" src="https://user-images.githubusercontent.com/5867739/228482246-e00324cf-756a-4e60-9056-6a6df7caf051.png">

`2023-03-28 22:58:06,076 - mmdet - ERROR - The testing results of the whole dataset is empty.`

(Notes: I double-checked the converted val.json by performing the inference of xView using your checkpoint on `TOOD+SF+SAHI+FI+PO` setup and I was able to get the announced result. Also, I was able to train xView with https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/xview_tood/tood_full_cls_60.py config that got low results when using inference with `TOOD+SF+SAHI+FI+PO` setup)

I'm trying to figure out what went wrong. Could you please help me with that?
Looking forward to your reply

**Comment by fcakyon** [comment]
Hello @nalemadi 

1. You should use the original training split during training (not the sliced dataset).
2. You can use the sliced val split during training.
3. You should use the [eval script](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/eval_tools/predict_evaluate_analyse.py) on the original dataset (not on the sliced dataset).

Is this what you are doing?

**Reply by nalemadi** [reply]
Yes, this is what I'm doing.
